### PR TITLE
Add auto-toolbox

### PR DESCRIPTION
**Pull request by gezakerecsenyi**

### CODE REVIEW

**Comment by palkerecsenyi** [comment]
Please provide a description.

File: dist/ffau.js [hunk 1]
```diff
@@ -2145,4 +2155,407 @@ class Ffau {
...
+                        return true;
+                } else {
+                    const newScope = Object.assign([], scopes);
+                    newScope.push(thisCategory.name);
+
+                    if (scopes.every(e => virtualScope.includes(e)) && scopes.length === virtualScope.length && thisCategory.name === category)
+                        return true;
+
+                    const recursiveRes = categoryExists(category, thisCategory.categories, virtualScope, newScope);
+                    if (recursiveRes)
```
Comment (palkerecsenyi) [inline dist/ffau.js:2183]: Please format this code properly

File: dist/ffau.js [hunk 2]
```diff
@@ -2145,4 +2155,407 @@ class Ffau {
...
+        if (blocks) {
+            let innerHTML = `<xml id="${id}" style="display: none;">`;
+
+            blocks.forEach(block => {
+                innerHTML += block === '*sep' ? '<sep></sep>' : `<block type="${block}"></block>`;
+            });
+
+            innerHTML += '</xml>';
+
+            if (parent)
```
Comment (palkerecsenyi) [inline dist/ffau.js:2200]: Please format this code properly

**Comment by gezakerecsenyi** [comment]
See https://github.com/codeddraig/ffau/blob/create_auto_toolbox/README.md line 89

File: dist/ffau.js [hunk 3]
```diff
@@ -2145,4 +2155,407 @@ class Ffau {
...
+     * @param {['all'] | FfauCategory[]} [categories]
+     * @param {'all' | string[]} [blocks]
+     */
+    renderToolbox(parent, id, categories, blocks) {
+        const categoryExists = (category, scope, virtualScope, scopes) => {
+            for (const thisCategory of (scope || categories)) {
+                if (typeof thisCategory === 'string') {
+                    if (scopes.every(e => virtualScope.includes(e)) &&
+                        (thisCategory === category || thisCategory === 'all')
+                    )
```
Comment (palkerecsenyi) [inline dist/ffau.js:2173]: Please format this code properly

**Comment by gezakerecsenyi** [comment]
(CTRL+F for 'Here's an example using all Ffau components and three blocks:')

File: dist/ffau.js [hunk 4]
```diff
@@ -2145,4 +2155,407 @@ class Ffau {
...
+                        return true;
+                } else {
+                    const newScope = Object.assign([], scopes);
+                    newScope.push(thisCategory.name);
+
+                    if (scopes.every(e => virtualScope.includes(e)) && scopes.length === virtualScope.length && thisCategory.name === category)
```
Comment (palkerecsenyi) [inline dist/ffau.js:2179]: Please format this code properly

Comment (gezakerecsenyi) [reply dist/ffau.js:2179]: It is formatted properly...

Comment (gezakerecsenyi) [reply dist/ffau.js]: OK done

**Comment by gezakerecsenyi** [comment]
@palkerecsenyi see now

**Pull request merged**